pkg/api/handlers/mcp.go

```diff
@@ -42,28 +43,63 @@ func NewMCPHandler(gptscript *gptscript.GPTScript, mcpLoader *mcp.SessionManager
 func (m *MCPHandler) GetCatalogEntry(req api.Context) error {
 	var (
 		entry v1.MCPServerCatalogEntry
-		id    = req.PathValue("mcp_server_id")
+		id    = req.PathValue("entry_id")
 	)
 
 	if err := req.Get(&entry, id); err != nil {
 		return err
 	}
 
+	// Authorization check.
+	if !req.UserIsAdmin() {
+		userCatalogAuthorizations, err := usercatalogauthorization.GetUserAuthorizationsForCatalog(req.Context(), req.Storage, req.Namespace(), req.User.GetUID(), entry.Spec.MCPCatalogName)
+		if err != nil {
+			return err
+		}
+
+		if len(userCatalogAuthorizations) == 0 {
+			return types.NewErrForbidden("user is not authorized to access this catalog entry")
+		}
+	}
+
 	return req.Write(convertMCPServerCatalogEntry(entry))
 }
 
-func (m *MCPHandler) ListCatalog(req api.Context) error {
+func (m *MCPHandler) ListEntriesForAllCatalogs(req api.Context) error {
 	var list v1.MCPServerCatalogEntryList
 	if err := req.List(&list); err != nil {
 		return err
 	}
 
-	items := make([]types.MCPServerCatalogEntry, 0, len(list.Items))
+	if req.UserIsAdmin() {
+		items := make([]types.MCPServerCatalogEntry, 0, len(list.Items))
+		for _, entry := range list.Items {
+			items = append(items, convertMCPServerCatalogEntry(entry))
+		}
+
+		return req.Write(types.MCPServerCatalogEntryList{Items: items})
+	}
+
+	userCatalogAuthorizations, err := usercatalogauthorization.GetAuthorizationsForUser(req.Context(), req.Storage, req.Namespace(), req.User.GetUID())
+	if err != nil {
+		return err
+	}
+
+	// TODO(g-linville): would it be better to do a separate list for each catalog that the user is authorized for,
```

Comment:
I think either is fine at this point, but once the set of catalogs/entries increases, we'll probably want to do separate lists.